### PR TITLE
Product page & basic routes

### PR DESCRIPTION
**Pull request by Manpreet-hub**
###  Products page and filter section for E-commerce. Below filters are added:- 
- filter slider by price.
- products rating
- products categories
- sorting products by price (low to high and high to low)
- Clear (removes all filters)

[Deployed link](https://ecomfashion.netlify.app/)

Screenshot (108)

### CODE REVIEW

File: src/components/Filters.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                <input id="women" type="checkbox" value="women" checked={category.includes("women")}  onClick={(e)=>{filterDispatch({type:"CATEGORY", payload: e.target.value})}}/>
+                <label htmlFor="women">Women Clothing</label>
+            </div>
+
+            <div className="category">
+                <input id="kids" type="checkbox" value="kids" checked={category.includes("kids")} onClick={(e)=>{filterDispatch({type:"CATEGORY", payload: e.target.value})}}/>
+                <label htmlFor="kids">Kids Clothing</label>
+            </div>
+        </div>
+
```
Comment (prathmeshjagtap) [inline src/components/Filters.jsx]: you  can make a array of category and map over it just like ratings

File: src/pages/Products.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import {  getCategoryFilter,getPricedFilter,getRatedFilter,getsortedProducts } from "../utils/filter-utils";
+
+const Products=()=>{
+  const {filterState}=useProductFilters();
+  const {sortBy,rating,price,category}=filterState;
+  const sortedProducts=getsortedProducts([...filterState.product],sortBy);
+  const ratedFilter=getRatedFilter(sortedProducts,rating);
+  const pricedFilter=getPricedFilter(ratedFilter,price);
+  const categoryFilter=getCategoryFilter(pricedFilter,category);
+
```
Comment (prathmeshjagtap) [inline src/pages/Products.jsx:13]: you can try making a compose function

File: src/styles/products.css [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+.product-img{
+    height: 70%;
+}
+
+.card{
+    background-color: white !important
+    
+}
```
Comment (prathmeshjagtap) [inline src/styles/products.css]: why are you using important here maybe try increasing specificity  by giving a addition card name

File: src/styles/products.css [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+.grid-container{
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(210px,1fr));
+    gap: 1.5rem;
+    /* padding: 1rem; */
```
Comment (prathmeshjagtap) [inline src/styles/products.css]: you can remove this

File: src/components/Filters.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        <div className="container-box">
+            <h3>Rating</h3>
+
+            {
+                [5, 4, 3, 2].map((rating,index)=>{
+                    return(
+                        <div className="rating-label" key={index}>
```
Comment (prathmeshjagtap) [inline src/components/Filters.jsx]: I think this should me 4,3,2,1 what if the user wants to see one star and above products , just a suggestion here

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]
Everything else looks fine, just try to implement the suggested changes

**Review by ajay-mandaviya — APPROVED** [review]
Everything looks great and well structure do change as some prathmeshjagtap suggest

File: src/components/Filters.jsx (hunk 5 above)
Comment (Manpreet-hub) [reply src/components/Filters.jsx]: Done [c89e2c3](https://github.com/Manpreet-hub/ecommerce-react/pull/2/commits/c89e2c395d0a67c4759fe952cd747cd645fe72d8)

File: src/styles/products.css (hunk 4 above)
Comment (Manpreet-hub) [reply src/styles/products.css]: Done [5b69145](https://github.com/Manpreet-hub/ecommerce-react/pull/2/commits/5b691456c4039caf5486bd9dd503c271e6193c2d)

File: src/styles/products.css (hunk 3 above)
Comment (Manpreet-hub) [reply src/styles/products.css]: Done [5b69145](https://github.com/Manpreet-hub/ecommerce-react/pull/2/commits/5b691456c4039caf5486bd9dd503c271e6193c2d)

File: src/components/Filters.jsx (hunk 1 above)
Comment (Manpreet-hub) [reply src/components/Filters.jsx]: Done [c89e2c3](https://github.com/Manpreet-hub/ecommerce-react/pull/2/commits/c89e2c395d0a67c4759fe952cd747cd645fe72d8)

File: src/pages/Products.jsx (hunk 2 above)
Comment (Manpreet-hub) [reply src/pages/Products.jsx:13]: Thanks! I shall do that while refactoring.

**Review by prathmeshjagtap — APPROVED** [review]

**Pull request merged**